src/modules/core/router/hooks/useRoutes.tsx

```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+import { Routes } from '../types';
+
+import RecipeListRoute from 'src/modules/recipe/modules/recipe-list/containers/RecipeListRoute';
+import RecipeDetailRoute from 'src/modules/recipe/modules/recipe-detail/containers/RecipeDetailRoute';
+import RecipeCreateRoute from 'src/modules/recipe/modules/recipe-create/containers/RecipeCreateRoute';
+import RecipeUpdateRoute from 'src/modules/recipe/modules/recipe-update/containers/RecipeUpdateRoute';
+
+export default function useRoutesParams(): Routes {
+  return React.useMemo(
```

Comment:
I also cannot see the point to use the `useMemo` here

Reply:
The point of that is DDD architecture. So, in an ideal scenario, we would have for example the route of `RecipeListRoute` at the same level as the container itself (`RecipeListRoute`). Then, in the core a helper called `getRoutes` which retrieves all the routes within the app and pass the to `RouterRoutes.tsx`. At the beginning, I was about to do that, but then I realized it will add to much complexity for this training exercise.